packages/core/upload/admin/src/utils/appendSearchParamsToUrl.js

```diff
@@ -6,15 +6,15 @@
  * @returns {String} A string representing the URL with the search params appended
  */
 const appendSearchParamsToUrl = ({ url, params }) => {
-  if (url === undefined || params === undefined || typeof params !== 'object' || Object.entries(params).length === 0) {
+  if (url === undefined || typeof params !== 'object') {
     return url;
   }
```

Comment:
Should we move this check outside the helper and only call it, when url is defined? Would make it even cleaner imo.

Reply:
i can't think of a scenario where url would not be defined, but my only case not to move it outside is that URL throws if you pass it undefined, so given our current issues, it might be safer to keep it inside... what do you think?

Reply:
I was thinking `appendSearchParamsToUrl` should add the passed params to the URL, so there is no point in calling it, if there is no URL, so I'd have catched that outside to make the util itself lighter. In case url is empty I'd instead throw an error.

But both ways are fine - it's just different styles of programming. I'll approve it.

Reply:
oh yes i see, i think that's really interesting, I think there's a lot of space in upload to do all this work, i'm quite excited to move to TS because we can really discuss this and refine naming / function scope etc. 🤩